tools/challenge-parser/parser/plugins/add-fill-in-the-blank.js

```diff
@@ -40,19 +42,86 @@ function plugin() {
     if (fillInTheBlankNodes.length > 0) {
       const fillInTheBlankTree = root(fillInTheBlankNodes);
 
-      validateBlanksCount(fillInTheBlankTree);
+      validateBlanksSectionCount(fillInTheBlankTree);
 
       const sentenceNodes = getSection(fillInTheBlankTree, '--sentence--');
       const blanksNodes = getSection(fillInTheBlankTree, '--blanks--');
 
-      const fillInTheBlank = getfillInTheBlank(sentenceNodes, blanksNodes);
-
-      file.data.fillInTheBlank = fillInTheBlank;
+      const lang = file.data.lang;
+      const inputType = file.data.inputType;
+      const toHtml = createMdastToHtml(lang);
+
+      file.data.fillInTheBlank = getFillInTheBlank(sentenceNodes, blanksNodes);
+
+      function getFillInTheBlank(sentenceNodes, blanksNodes) {
+        const sentenceWithoutCodeBlocks = sentenceNodes.map(node => {
+          node.children.forEach(child => {
+            if (child.type === 'text' && child.value.trim() === '')
+              throw Error(NOT_IN_PARAGRAPHS);
+            if (child.type !== 'inlineCode') throw Error(NOT_IN_CODE_BLOCK);
+          });
+          const children = node.children.map(child => ({
+            ...child,
+            type: 'text'
+          }));
+          return { ...node, children };
+        });
+        const sentence = toHtml(sentenceWithoutCodeBlocks);
+        const blanks = getBlanks(blanksNodes);
+
+        if (!sentence)
+          throw Error('sentence is missing from fill in the blank');
+        if (!blanks) throw Error('blanks are missing from fill in the blank');
+
+        validateBlankCount(sentenceNodes, blanks);
```

Comment:
... this.

I moved this check:

https://github.com/freeCodeCamp/freeCodeCamp/blob/3d7dea735baa070c3e36af7c87d2c8e1eb849c74/tools/challenge-parser/parser/plugins/add-fill-in-the-blank.js#L83-L86

into the `validateBlankCount` function. 

I'll explain the function in a comment below.